packages/react-jss/src/withStyles.js

```diff
@@ -0,0 +1,259 @@
+// @flow
+import React, {Component, type ComponentType, type Node} from 'react'
```

Comment:
Why? If its just taste? I prefer to always use specific named imports instead of an object that has everything? Shouldn't this  get us some optimizations once react has esm bundle?

Reply:
Nope. Rollup expands namespaces with usages, so in the end you see the same expanded set of named imports.
With default export you get a bloat with whole react api object.
And named exports are more natural in context of react stuff. Default exports is again a legacy of cjs.

Reply:
Can you make an example please? I didn't get it.

Reply:
```js
import * as React from 'react';

React.Component;
React.createContext;
```
is expanded into
```js
import {Component, createContext} from 'react';

Component;
createContext;
```

This is super nice optimisation feature of rollup. Webpack don't do this and instead duplicates `React.createElement` everywhere which increase bundle size a lot. Guys invents babel/webpack plugins for this.

For me it's just absurd. Webpack makes me said last months. It has fast enough development mode but production is far from perfect.

Reply:
So what speaks against manually writing `import {Component, createContext} from 'react';`?

Reply:
I personally don't use default export. But I ask only to stop import react as default to reduce bundle size and simplify future transition.

Reply:
What exactly is the impact in bundle size when default export is used?

Reply:
`React.default.createElement` instead of just `createElement` for example. Doesn't solve the problem for webpack though.

Reply:
This size is not going away with uglify?